Remove the precision="float" attribute on <input type="range">

Ahmad-S792 opened the pull request:
#### 06479587e171c15e02394bc40840619744826116
<pre>
Remove the precision=&quot;float&quot; attribute on &lt;input type=&quot;range&quot;&gt;

Remove the precision=&quot;float&quot; attribute on &lt;input type=&quot;range&quot;&gt;
<a href="https://bugs.webkit.org/show_bug.cgi?id=250047">https://bugs.webkit.org/show_bug.cgi?id=250047</a>

Reviewed by Alexey Shvayka.

This patch is to align WebKit with Gecko / Firefox, Blink / Chromium by removing non-standard &apos;precision&apos; attribute.

Merge - <a href="https://chromium.googlesource.com/chromium/blink/+/6b76e9c0e649bf8d4b1c6c321af513e0e0c6c1af">https://chromium.googlesource.com/chromium/blink/+/6b76e9c0e649bf8d4b1c6c321af513e0e0c6c1af</a>

This was added in WebKit but it was non-standard and it was never added by any other browser.
Blink removed it in 2014 so only WebKit support it as of right now and it is broken feature
since it does not support &apos;step&apos;  with precision=&quot;float&quot;.

* Source/WebCore/html/HTMLAttributeNames.in: Remove &quot;precision&quot;
* Source/WebCore/html/HTMLInputElement.cpp: Remove &quot;precisionAttr&quot;
* Source/WebCore/html/RangeInputType.cpp: Remove &quot;precision=&apos;float&apos;&quot; support and FIXME

Canonical link: <a href="https://commits.webkit.org/258625@main">https://commits.webkit.org/258625@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9da350e026640c10c63521e093b4aa169662e4fc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102508 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11642 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35542 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/111778 "Built successfully") | [💥 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/171984 "An unexpected error occured. Recent messages:Pull request contains relevant changes; Failed to print configuration") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106477 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12649 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2543 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/94786 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/109489 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108288 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/9697 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/92920 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37349 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91559 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24409 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/79099 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5100 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/25829 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5261 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2279 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11279 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45320 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5927 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/6993 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

shvaikalesh commented on Source/WebCore/html/RangeInputType.cpp:
```diff
@@ -1,6 +1,6 @@
 /*
- * Copyright (C) 2010 Google Inc. All rights reserved.
- * Copyright (C) 2011-2018 Apple Inc. All rights reserved.
+ * Copyright (C) 2010-2014 Google Inc. All rights reserved.
```
Haven't Blink forked in 2013?

Ahmad-S792 replied:
Yes - I update Copyright year based on the year of patch, we are merging, since it was in 2014, so I update it to 2014. I checked this with @rniwa in the past. :-)

shvaikalesh reviewed (approved):
Lovely!

Ahmad-S792 committed:
Remove the precision="float" attribute on <input type="range">

Remove the precision="float" attribute on <input type="range">
https://bugs.webkit.org/show_bug.cgi?id=250047

Reviewed by Alexey Shvayka.

This patch is to align WebKit with Gecko / Firefox, Blink / Chromium by removing non-standard 'precision' attribute.

Merge - https://chromium.googlesource.com/chromium/blink/+/6b76e9c0e649bf8d4b1c6c321af513e0e0c6c1af

This was added in WebKit but it was non-standard and it was never added by any other browser.
Blink removed it in 2014 so only WebKit support it as of right now and it is broken feature
since it does not support 'step'  with precision="float".

* Source/WebCore/html/HTMLAttributeNames.in: Remove "precision"
* Source/WebCore/html/HTMLInputElement.cpp: Remove "precisionAttr"
* Source/WebCore/html/RangeInputType.cpp: Remove "precision='float'" support and FIXME

Canonical link: https://commits.webkit.org/258625@main
